libs/features/jarvis-chat.js

```diff
@@ -0,0 +1,343 @@
+  chatWorkflow: data['event.workflow'],
+  chatCategory: data['event.category'],
+  chatSubcategory: data['event.subcategory'],
+  chatType: data['event.type'],
+  chatSubtype: data['event.subtype'],
+  chatUserguid: data['event.user_guid'],
+  chatLanguage: data['event.language'],
+  chatPagename: data['event.pagename'],
+  chatVisitorguid: data['event.visitor_guid'],
+  chatUrl: data['event.url'],
+  chatOrgguid: data['event.org_guid'],
+  chatReferrer: data['event.referrer'],
+  chatClientId: data['source.client_id'],
+  chatSourceName: data['source.name'],
+  chatPlatform: data['source.platform'],
+  chatContentName: data['content.name'],
+  chatEnv: data['env.com.name'],
+});
+
+const setSophiaLayer = (data) => {
+  const sophiaData = {
+    variationId: data['exp.variation_id'],
+    actionBlockId: data['exp.action_block_id'],
+    campaignId: data['exp.campaign_id'],
+    containerId: data['exp.container_id'],
+    controlGroupId: data['exp.control_group_id'],
+    treatmentId: data['exp.treatment_id'],
+    surfaceId: data['exp.surface_id'],
+  };
+
+  Object.keys(sophiaData).forEach((key) => {
+    if (!sophiaData[key]) delete sophiaData[key];
+  });
+  if (!Object.keys(sophiaData).length) return;
+
+  const sophiaDataCaptured = (id) => {
+    let captured = false;
+    if (!id) return captured;
+
+    const sophiaDataLayer = window.digitalData.sophiaResponse?.fromPage;
+    if (Array.isArray(sophiaDataLayer)) {
+      captured = sophiaDataLayer.some((sophiaEvent) => sophiaEvent?.campaignId === id);
+    } else {
+      captured = (sophiaDataLayer?.campaignId === id);
+    }
+
+    return captured;
+  };
+
+  if (sophiaDataCaptured(sophiaData.campaignId)) return;
+
+  window.digitalData.sophiaResponse = window.digitalData.sophiaResponse || {};
+  window.digitalData.sophiaResponse.fromPage = window.digitalData.sophiaResponse.fromPage || [];
+
+  if (Array.isArray(window.digitalData.sophiaResponse.fromPage)) {
+    window.digitalData.sophiaResponse.fromPage.push(sophiaData);
+  } else {
+    window.digitalData.sophiaResponse.fromPage = sophiaData;
+  }
+};
+
+const sendEvent = (data) => {
+  setSophiaLayer(data);
+  // eslint-disable-next-line no-underscore-dangle
+  window._satellite?.track('event', {
+    xdm: {},
+    data: {
+      web: { webInteraction: { name: window.digitalData.primaryEvent?.eventInfo?.eventName } },
+      _adobe_corpnew: {
+        digitalData: {
+          primaryEvent: window.digitalData.primaryEvent,
+          chat: window.digitalData.chat,
+          sophiaResponse: window.digitalData.sophiaResponse,
+        },
+      },
+    },
+  });
+};
+
+const sendChatIconRenderEvent = (data) => {
+  window.digitalData.primaryEvent = {
+    eventInfo: {
+      eventName: 'chat:'
+        + `${data['event.workflow'] || ''}:`
+        + `${data['event.subcategory'] || ''}:`
+        + `${data['event.subtype'] || ''}:`
+        + `${data['content.name'] || ''}:`
+        + `${data['event.type'] || ''}`,
```

Comment:
Since this is used multiple times, maybe use a fn that can do the aggregation?

something like:
```
const getEventStr = (data, props = []) => 
  props.reduce((str, prop) => {
    str += `${str && ':'}${data[prop] || ''}`;
    return str;
  }, '');
```

usage:
```
eventName: `chat:${getEventStr(data, ['event.workflow', 'event.subcategory', 'event.subtype', ...])}`
```